### ISSUE TITLE
DefaultBindingFactory creates environment with wrong PropertySources.

### ISSUE THREAD

**Issue by tan9** [spring-cloud/spring-cloud-stream#706]
I have a project uses `@PropertySource` to provide default binder properties, and ought to be overridden by an `application.yml` file.

However, I noticed that `DefaultBindingFactory` created ApplicationContext has propertySources in wrong order like:

```
[0] commandLineArgs,
[1] systemProperties,
[2] systemEnvironment,
[3] random,
[4] servletConfigInitParams,
[5] servletContextInitParams,
[6] class path resource [kafka-default.properties],
[7] Management Server,
[8] applicationConfig: [classpath:/application.yml]
```

than in the parent applictionContext:
```
[0] servletConfigInitParams,
[1] servletContextInitParams,
[2] systemProperties,
[3] systemEnvironment,
[4] random,
[5] applicationConfig: [classpath:/application.yml],
[6] class path resource [kafka-default.properties]]
```

So that one can not override any `@PropertySource` properties via `application.yml` in binder, and system properties takes precedence over Servletonfig init params.

----
A simple project demonstrating this behavior can be found here:
https://github.com/tan9/issue-spring-cloud-stream-binder-factory

You can set break point on `KafkaBinderConfigurationProperties#getZkConnectionString` to observe the differences between plain Sprig Boot configuration properties and DefaultBinderFactory.

**Comment by mbogoevici** [comment]
Thanks for reporting @tan9 - if you'd like to take a stab at a fix, we welcome contributions :)

**Comment by tan9** [comment]
@mbogoevici I, am willing to contribute, but don't know how to do it correctly.
Should it be done in Spring Boot side, or we can postprocess the ApplicatuonContext created by Spring Boot in DefaultBindingFactory?

**Comment by sobychacko** [comment]
Hello @tan9 - apologies for the long delay on this issue. We are taking a look at this right now and realized that a lot of things have changed since the issue opened originally. Could you please update to the latest spring-boot (2.0 RC2 or snapshot) /spring-cloud-stream (2.0.0.M4 or snapshot) to verify if this is still an issue (or not)? If it still is, please ping here and we will try to fix it in 2.0.0.RC2. Otherwise, we will close the issue. Thanks!

**Comment by tan9** [comment]
@sobychacko thanks for the information, I am going to close this issue right now. And I will create new issue if we have any chance trying it out in the latest release.
